Add guidelines for image comparison tests

Wumpf opened the pull request:
Guidelines & why images may differ

Based on (but slightly altered):
* https://github.com/rerun-io/rerun/pull/8989

Wumpf commented on crates/egui_kittest/README.md:
```diff
@@ -55,3 +55,48 @@ You should add the following to your `.gitignore`:
+### Guidelines for writing snapshot tests
+
+* Whenever **possible** prefer regular Rust tests or `insta` snapshot tests over image comparison tests because…
+  * …they are slow to run
```
```suggestion
  * …they are relatively slow to run
```

Wumpf commented on crates/egui_kittest/README.md:
```diff
@@ -55,3 +55,48 @@ You should add the following to your `.gitignore`:
+  * …they are brittle since unrelated side effects (like a change in color) can cause the test to fail
+  * …images take up repo space
+* images should…
+  * …be checked in as LFS file
```
that one is more specific to egui & rerun, should write something else here

lucasmerlin replied:
Yeah I think we can suggest using lfs, but for small projects with only a couple of snapshot tests it's probably fine to just have them in git

lucasmerlin commented on crates/egui_kittest/README.md:
```diff
@@ -55,3 +55,48 @@ You should add the following to your `.gitignore`:
+  * …be checked in as LFS file
+  * …depict exactly what's tested and nothing else
+  * …have a low resolution to avoid growth in repo size
+  * …have a low comparison threshold to avoid the test passing despite unwanted differences
```
```suggestion
  * …have a low comparison threshold to avoid the test passing despite unwanted differences (the default treshold should be fine)
```

lucasmerlin reviewed (commented):
This is great, but I'm a bit worried that the "Why does CI / another computer produce a different image?" section might overwhelm the average user who just wants to test whether their button still looks the same. 
If they run into issues with snapshots being different in CI, the reasonable thing for them to do would be increasing the threshold until it works. I think we should mention that somewhere, before explaining all the rendering things.

Wumpf commented:
I disagree, it's important to understand why an image comparison tests may fail or even be flaky if you ever run into that situation. I do recognize that the detail I went into here stretches the intended audience by quite a bit which may only be concerned with "simple" ui testing. But as I see it it offers a way for more systematic reasoning than just mindlessly adjusting thresholds until it works (which, while being the inevitable reality 😞, is obviously a very precarious practice).
Not feeling super strongly about this though, I can always just leave that in in the Rerun contribution docs only 🤷  (Emil [suggested](https://github.com/rerun-io/rerun/pull/8989#discussion_r1952101726) to put this here and I thought that made sense)

Wumpf commented:
maybe I can adjust the text a bit to make this more of "stop, think, look at the changes" before going into the weeds.

Wumpf commented:
I changed the framing and added more words talking about immediate actions instead. What do you think @lucasmerlin , does this hit the right compromise or would you still rather leave that out entirely?

lucasmerlin reviewed (approved):
Awesome, I think it's perfect now!
